### PR TITLE
ASC-1253 Automatically Extract OpenStack Properties from Ansible

### PR DESCRIPTION
**Pull request by ryan-rs**
Update the "openstack_properties" fixture to automatically extract all the
variables from Ansible. Also, I update the naming of the Ansible variables
to be consistent. Let me know if the naming is confusing.

### CODE REVIEW

**Comment by ryan-rs** [comment]
@johnduarte @phongdly Updated the PR with changes. I found a bunch of other things that needed to be changed. Also, I shortened the fixture name to avoid having to make wild adjustments to fit within 80 columns.

File: vars/main.yml [hunk 1]
```diff
@@ -26,11 +26,15 @@ image:
...
-test_router: TEST-ROUTER
-test_network: TEST-VXLAN
-test_subnet: TEST-VXLAN-SUBNET
+test_resources:
+  server: post-deploy-server
+  volume: post-deploy-volume
+  router: TEST-ROUTER
+  network: TEST-VXLAN
+  subnet: TEST-VXLAN-SUBNET
+  image_name: Cirros-0.3.5
```
Comment (johnduarte) [inline vars/main.yml]: I would not consider this a member of `test_resources`. My definition of `test_resources` are those resources create during converge after the completion of the `openstack-ansible-ops` playbook. Such resources are created just for the purposes of convenience. These resource will be removed from the converge step in the future.

The "Cirros-0.3.5" image is created by the `openstack-ansible-ops` playbook. Given our current validation, this is considered to be part of a base product installation.

This key/value should be moved out the the `test_resources` namespace.

File: vars/main.yml [hunk 2]
```diff
@@ -26,11 +26,15 @@ image:
...
+  flavor: m1.tiny
```
Comment (johnduarte) [inline vars/main.yml]: See note above.

The "m1.tiny" flavor is created by the `openstack-ansible-ops` playbook. Given our current validation, this is considered to be part of a base product installation.

This key/value should be moved out the the `test_resources` namespace.

File: vars/main.yml [hunk 3]
```diff
@@ -26,11 +26,15 @@ image:
...
+  zone: nova
```
Comment (johnduarte) [inline vars/main.yml]: See not above.

The "nova" zone is created as part of a base product installation.

This key/value should be moved out the the `test_resources` namespace.

File: molecule/default/tests/test_instance_per_network_per_hypervisor.py [hunk 4]
```diff
@@ -43,18 +43,23 @@ def create_server_on(target_host,
...
 
     ssh = "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
            -i ~/.ssh/rpc_support ubuntu@{}"
 
-    vars = host.ansible('include_vars',
-                        'file=./vars/main.yml')['ansible_facts']
-
-    flavor_name = vars['flavor']['name']
-    image_name = vars['image']['name']
+    flavor_name = os_props['flavor']['name']
```
Comment (johnduarte) [inline molecule/default/tests/test_instance_per_network_per_hypervisor.py]: This flavor is one created for test purposes ("post-deploy").  It should be moved into the `test_resources` dictionary.

File: molecule/default/tests/test_instance_per_network_per_hypervisor.py [hunk 5]
```diff
@@ -43,18 +43,23 @@ def create_server_on(target_host,
...
+    image_name = os_props['image']['name']
```
Comment (johnduarte) [inline molecule/default/tests/test_instance_per_network_per_hypervisor.py]: This is also an arbitrary image used for creating test instances. I am fine leaving this as is, but it probably needs a tech debt ticket to ensure that we have a better way in the future of defining and using these values. Our tests currently use both the "Ubuntu 16.04" and "Cirros-0.3.5" images for instance creation and validation.

**Comment by ryan-rs** [comment]
@johnduarte Updated again. I just went ahead and moved the `flavor` and `image` under `test_resources` to keep things organized.

**Pull request merged**